### PR TITLE
NAS-120647 / 23.10 / Add some casefolded filters

### PR DESCRIPTION
**Pull request by anodos325**
Add special behavior where capital C prefix in filterop performs a casefolded version of
the filter. This is useful in cases where we're dealing with external protocols that are
case-insensitive, and will hopefully lead to fewer bugs related to case-insensitive checks.

### CODE REVIEW

**Comment by anodos325** [comment]
Throwing WIP back on because of some CI failures.

**Comment by anodos325** [comment]
Last custom jenkins run came up clean.

File: src/middlewared/middlewared/utils/__init__.py [hunk 1]
```diff
@@ -111,8 +111,33 @@ def get(obj, path):
...
+
+    if isinstance(obj, (list, tuple)):
+        return [x.casefold() for x in obj]
+
+    raise ValueError(f'{type(obj)}: support for casefolding object type not implemented.')
+
+
 class filters(object):
-    opmap = {
+    case_opmap = {
```
Comment (themylogin) [inline src/middlewared/middlewared/utils/__init__.py:128]: Can we just use operators from `opmap` but run `casefold` against arguments first? That way we won't have duplicate code.

**Review by themylogin — CHANGES_REQUESTED** [review]
Also `casefold` seems confusing here, like there is some reason to perform indeed `casefold` and not, for example, just `lower`

**Review by themylogin — APPROVED** [review]

**Pull request merged**